src/Features/Core/Portable/MetadataAsSource/DecompilationMetadataAsSourceFileProvider.cs

```diff
@@ -172,13 +195,16 @@ public DecompilationMetadataAsSourceFileProvider(IImplementationAssemblyLookupSe
                     }
                 }
 
-                using (var textWriter = new StreamWriter(fileInfo.TemporaryFilePath, append: false, encoding: MetadataAsSourceGeneratedFileInfo.Encoding))
+                if (!skipWritingFile)
```

Comment:
What's the consequence of not creating the file? We'd simply try again the next time it was request?

Reply:
That seems to be the case. Also, the compilation that gets created for it will see an empty file at that location with an error about failing to load the file contents.

Reply:
Is it possible to show the golden bar here? Or just maybe log the `DirectoryNotFoundException` or `UnauthorizedAccessException` so we have some traces of what's going on here

Reply:
Looks like my question is asked in the preview comment : )

Reply:
We _should_ still get that. I'm not sure how to force this situation locally so it's hard to test.